### PR TITLE
feat: [#185649292] add navbar business name logic for OWNING businesses

### PR DESCRIPTION
**Pull request by aloverso**
### Ticket

[#185649292](https://www.pivotaltracker.com/story/show/#185649292)

### Approach

Reorganized test code to make it easier to understand the different cases that cause different outputs

Rewrote prod code to support Oscar

### Notes

Interested in feedback on the readability of this test reorganization

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation, if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config

### CODE REVIEW

**Review by cbrunefearless — APPROVED** [review]
lgtm

File: web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx [hunk 1]
```diff
@@ -1,141 +1,344 @@
...
+            tradeName: name,
+            legalStructureId: undefined,
+          }),
+        });
+        const navBarBusinessTitle = getNavBarBusinessTitle(business);
+        expect(navBarBusinessTitle).toEqual(name);
+      });
+
+      it.each(["STARTING", "OWNING"])(
+        "shows business name over trade name if both defined",
```
Comment (somabadri) [inline web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:60]: When would this situation occur in our application?

File: web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx [hunk 2]
```diff
@@ -1,141 +1,344 @@
...
   });
 
-  it("shows unnamed[industryId name][legalStructureId] abbreviation if legalStructureId and industryId are defined and trade name is undefined", () => {
-    const business = generateBusiness({
-      profileData: generateProfileData({
-        tradeName: "",
-        industryId: "cannabis",
-        legalStructureId: "sole-proprietorship",
-      }),
+  describe("when legal structure, industry, name undefined", () => {
```
Comment (somabadri) [inline web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:271]: [Dust] , and name

File: web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx [hunk 3]
```diff
@@ -1,141 +1,344 @@
...
-        tradeName: "",
-        industryId: undefined,
-      }),
-    });
-    const navBarBusinessTitle = getNavBarBusinessTitle(business);
-    expect(navBarBusinessTitle).toEqual(Config.navigationDefaults.navBarGuestText);
-  });
+  describe("when name is defined", () => {
+    describe("when legal structure undefined", () => {
+      it.each(["STARTING", "OWNING"])("shows business name", (businessPersona) => {
```
Comment (somabadri) [inline web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:33]: This is a cool pattern, just to confirm it's an alternative to a loop right.

**Review by somabadri — APPROVED** [review]
Looks good to me, just left a few questions and one comment but ready to be merged!

File: web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx (hunk 1 above)
Comment (aloverso) [reply web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:60]: If a user switches legal structure. It's not likely but technically possible
1. onboard 
2. set LLC
3. save a business name in profile
4. switch to SP/GP
5. save a trade name in profile
6. remove legal structure altogether

File: web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx (hunk 3 above)
Comment (aloverso) [reply web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:33]: yeah @adidner did a demo on `it.each` a while back, I think it's all over our codebase now

Comment (somabadri) [reply web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx:33]: oh yeah, when he first joined the team. This is the first time I'm seeing it but yeah it is throughout. ty for the clarification.

**Pull request merged**